standalone-metastore/metastore-common/src/test/java/org/apache/hadoop/hive/common/metrics/metrics2/TestCodahaleReportersConf.java

```diff
@@ -149,4 +151,31 @@ public void testNoFallbackOnIncorrectConf() throws Exception {
 
     Assert.assertFalse(jsonReportFile.exists());
   }
+
+  /**
+   * Tests if MetricsFactory is initialised properly, if in case Metrics2Reporter is tried to add more than once.
+   *
+   * **/
+  @Test
+  public void testMetricsFactoryInitMetrics2ReporterAddedTwice() throws Exception {
+    if(MetricsFactory.getInstance() != null) {
```

Comment:
why is this needed? I can see an @After method taking care of this
but if it turns out this is needed, you can still add it as a @Before method (to make it applied before every test method)

Reply:
Okay I will refactor the code